src/thd_dbus_interface.cpp

```diff
@@ -580,6 +580,670 @@ gboolean thd_dbus_interface_get_sensor_temperature(PrefObject *obj, int index,
+			return;
+		}
+		return;
+	}
+
+	if (g_strcmp0(method_name, "SetUserPassiveTemperature") == 0) {
+		gboolean ret;
+		g_autofree gchar *zone_name = NULL;
+		guint user_set_point_in_milli_degree_celsius;
+
+		g_variant_get(parameters, "(su)", &zone_name,
+			      &user_set_point_in_milli_degree_celsius);
+
+		ret = thd_dbus_interface_set_user_passive_temperature(obj, zone_name,
+								      user_set_point_in_milli_degree_celsius,
+								      &error);
+		if (!ret) {
+			g_dbus_method_invocation_return_gerror(invocation, error);
+			return;
+		}
+
+		g_dbus_method_invocation_return_value(invocation, NULL);
+		return;
+	}
+
+	if (g_strcmp0(method_name, "SetZoneStatus") == 0) {
+		gboolean ret;
+		g_autofree gchar *zone_name = NULL;
+		gint status;
+
+		g_variant_get(parameters, "(si)", &zone_name, &status);
+
+		ret = thd_dbus_interface_set_zone_status(obj, zone_name, status,
+							 &error);
+
+		if (!ret) {
+			g_dbus_method_invocation_return_gerror(invocation, error);
+			return;
+		}
+
+		g_dbus_method_invocation_return_value(invocation, NULL);
+		return;
+	}
+
+	if (g_strcmp0(method_name, "Terminate") == 0) {
+		g_dbus_method_invocation_return_value(invocation, NULL);
+		thd_dbus_interface_terminate(obj, &error);
+		return;
+	}
+
+	if (g_strcmp0(method_name, "UpdateCoolingDevice") == 0) {
+		gboolean ret;
+		g_autofree gchar *cdev_name = NULL;
+		g_autofree gchar *path = NULL;
+		gint min_state;
+		gint max_state;
+		gint step;
+
+		g_variant_get(parameters, "(ssiii)", &cdev_name, &path, &min_state,
+			      &max_state, &step);
+
+		ret = thd_dbus_interface_update_cooling_device(obj,
+							       cdev_name, path, min_state, max_state,
+							       step, &error);
+		
+		if (!ret) {
+			g_dbus_method_invocation_return_gerror(invocation, error);
+			return;
+		}
+
+		g_dbus_method_invocation_return_value(invocation, NULL);
+		return;
+	}
+
+	g_set_error(&error,
+		    G_DBUS_ERROR,
+		    G_DBUS_ERROR_UNKNOWN_METHOD,
+		    "no such method %s",
+		    method_name);
+	g_dbus_method_invocation_return_gerror(invocation, error);
+}
+
+static GVariant *
+thd_dbus_handle_get_property(GDBusConnection  *connection,
+			     const gchar      *sender,
+			     const gchar      *object_path,
+			     const gchar      *interface_name,
+			     const gchar      *property_name,
+			     GError          **error,
+			     gpointer          user_data)
+{
+	return NULL;
+}
+
+static gboolean
+thd_dbus_handle_set_property(GDBusConnection  *connection,
+			     const gchar      *sender,
+			     const gchar      *object_path,
+			     const gchar      *interface_name,
+			     const gchar      *property_name,
+			     GVariant         *value,
+			     GError          **error,
+			     gpointer          user_data) {
+	return TRUE;
+}
+
+
+static void
+thd_dbus_on_bus_acquired(GDBusConnection *connection,
+			 const gchar     *name,
+			 gpointer         user_data) {
+	guint registration_id;
+	GDBusProxy *proxy_id = NULL;
+	GError *error;
```

Comment:
should we set error to NULL?
according to https://docs.gtk.org/gio/func.resources_lookup_data.html,
"
The argument can be NULL.
If the return location is not NULL, then you must initialize it to a NULL GError*.
The argument will left initialized to NULL by the function if there are no errors.
In case of error, the argument will be set to a newly allocated GError; the caller will take ownership of the data, and be responsible for freeing it."

Reply:
It should be NULL. I'll fix this.